plugins/lua/lua.c

```diff
@@ -1262,36 +1262,31 @@ static void prepare_state(lua_State *L, script_info *info)
 	lua_pop(L, 1);
 }
 
-struct unload_userdata {
-	lua_State *L;
-	int base;
-};
-
-static void run_unload_hook(hook_info *hook, struct unload_userdata *ud)
+static void run_unload_hook(hook_info *hook, lua_State *L)
 {
-	lua_State *L = ud->L;
-	int base = ud->base;
+	int base = lua_gettop(L);
 
 	lua_rawgeti(L, LUA_REGISTRYINDEX, hook->ref);
 	if(lua_pcall(L, 0, 0, base))
 	{
 		char const *error = lua_tostring(L, -1);
-		lua_pop(L, 2);
 		hexchat_printf(ph, "Lua error in unload hook: %s", error ? error : "(non-string error)");
 	}
+	lua_settop(L, base);
+}
+
+static void run_unload_hooks(script_info *info, void *unused)
```

Comment:
`/* The unused parameter is present because this function is used as a GFunc */`